src/types/errors.rs

```diff
@@ -0,0 +1,9 @@
+#[non_exhaustive]
+#[derive(Debug)]
+pub enum KafkaErrorCode {
```

Comment:
I would like to do away with this entirely. If we cannot for some reason, can we keep it inside the `kafka` implementation as it seems very kafka specific.

Reply:
I think this is actually the point. I think we can make arroyo agnostic to the specific Kafka library we use under the hoods, but we cannot really make it fully Kafka protocol agnostic. TheKafka protocol impacts the way the arroyo consumer interface works (think about callbacks). If you look at an AWS Kinesis protocol instead, it is quite different. It is not just function names. 

So in a world where our interface is the Kafka protocol (which means hiding the library and the specific Kafka protocol implementation (which incldues Redpanda) I don't see an issue with preserving the Kafka error codes in our interface.

Whether or not we expose them as error codes or as individual entires in the abstract `ConsumerError` enum is a different issue. In that case we could have all 99 entires in the `ConsumerError`. I feel that grouping them and preserving the error codes is a simpler solution for the client as they can refer to the Kafka protocol error code instead of having to map them to the arroyo errors. What do you think ?

Reply:
I think we should not have any ConsumerError::OffsetError, which contains one of these codes but actually just make separate ConsumerError members for some (or all) or these. i.e. Instead of `ConsumerError::OffsetError`, I think we should have `ConsumerError::UnknownTopicOrPartition` and `ConsumerError::InvalidMessage`, etc. Those errors are less Kafka specific.

Reply:
Works for me. Yes, we can revisit later. Either way is fine with me for this PR.